src/app/code/community/Zendesk/Zendesk/Model/Resource/Tickets/Collection.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/**
+ * Copyright 2015 Zendesk
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+class Zendesk_Zendesk_Model_Resource_Tickets_Collection extends Varien_Data_Collection {
+
+    protected $_count;
+    protected $_search;
+    protected $_viewColumns = array();
+    protected static $_excludedColumns = array('score');
+
+    public function __construct() {
+        $this->_search = new Zendesk_Zendesk_Model_Search( Zendesk_Zendesk_Model_Search::TYPE_TICKET );
+    }
+
+    public function addFieldToFilter($fieldName, $condition = null) {
+        if(is_string($condition) OR is_array($condition)) {
+            
+            switch($fieldName) {
+                case 'subject':
```

Comment:
This conditional would be a lot cleaner if it only called `$this->_search->addField( new Zendesk_Zendesk_Model_Search_Field( ));` once at the end of the function. What about if you extracted this into 2 logical blocks: build the search field value, then add it to the field. Right now it's very messy.